[TZone] WebCore/page Convert FastMalloc to TZone

msaboff opened the pull request:
#### f3f4594c841728632df8cf83e85566df2b987ea7
<pre>
[TZone] WebCore/page Convert FastMalloc to TZone
<a href="https://bugs.webkit.org/show_bug.cgi?id=278419">https://bugs.webkit.org/show_bug.cgi?id=278419</a>
<a href="https://rdar.apple.com/134372652">rdar://134372652</a>

Reviewed by Keith Miller.

Convert WTF_MAKE_FAST_ALLOCATED to WTF_MAKE_TZONE_ALLOCATED in
preparation for enabling TZone (not yet enabled).

* Source/WebCore/page/AlternativeTextClient.h:
* Source/WebCore/page/AttachmentElementClient.h:
* Source/WebCore/page/AudioOutputProvider.h:
* Source/WebCore/page/AutoscrollController.cpp:
* Source/WebCore/page/AutoscrollController.h:
* Source/WebCore/page/CaptionUserPreferences.cpp:
* Source/WebCore/page/CaptionUserPreferences.h:
* Source/WebCore/page/CaptionUserPreferencesMediaAF.cpp:
* Source/WebCore/page/CaptionUserPreferencesMediaAF.h:
* Source/WebCore/page/ContextMenuController.cpp:
* Source/WebCore/page/ContextMenuController.h:
* Source/WebCore/page/CryptoClient.h:
* Source/WebCore/page/DOMTimer.cpp:
* Source/WebCore/page/DOMTimer.h:
* Source/WebCore/page/DeviceClient.h:
* Source/WebCore/page/DeviceController.cpp:
* Source/WebCore/page/DeviceController.h:
* Source/WebCore/page/DiagnosticLoggingClient.h:
* Source/WebCore/page/DragClient.h:
* Source/WebCore/page/DragController.cpp:
* Source/WebCore/page/DragController.h:
* Source/WebCore/page/EditorClient.h:
* Source/WebCore/page/ElementTargetingController.cpp:
* Source/WebCore/page/ElementTargetingController.h:
* Source/WebCore/page/EmptyAttachmentElementClient.h:
* Source/WebCore/page/EventHandler.cpp:
* Source/WebCore/page/EventHandler.h:
* Source/WebCore/page/FocusController.cpp:
* Source/WebCore/page/FocusController.h:
* Source/WebCore/page/FrameView.cpp:
* Source/WebCore/page/FrameView.h:
* Source/WebCore/page/ImageAnalysisQueue.cpp:
* Source/WebCore/page/ImageAnalysisQueue.h:
* Source/WebCore/page/ImageOverlayController.cpp:
* Source/WebCore/page/ImageOverlayController.h:
* Source/WebCore/page/LocalFrameViewLayoutContext.cpp:
* Source/WebCore/page/LocalFrameViewLayoutContext.h:
* Source/WebCore/page/NavigatorLoginStatus.cpp:
* Source/WebCore/page/NavigatorLoginStatus.h:
* Source/WebCore/page/Page.cpp:
* Source/WebCore/page/Page.h:
* Source/WebCore/page/PageConfiguration.cpp:
* Source/WebCore/page/PageConfiguration.h:
* Source/WebCore/page/PageConsoleClient.cpp:
* Source/WebCore/page/PageConsoleClient.h:
* Source/WebCore/page/PageDebuggable.cpp:
* Source/WebCore/page/PageDebuggable.h:
* Source/WebCore/page/PageGroup.cpp:
* Source/WebCore/page/PageGroup.h:
* Source/WebCore/page/PageOverlay.cpp:
* Source/WebCore/page/PageOverlay.h:
* Source/WebCore/page/PageOverlayController.cpp:
* Source/WebCore/page/PageOverlayController.h:
* Source/WebCore/page/PagePasteboardContext.h:
* Source/WebCore/page/PerformanceLogging.cpp:
* Source/WebCore/page/PerformanceLogging.h:
* Source/WebCore/page/PerformanceLoggingClient.cpp:
* Source/WebCore/page/PerformanceLoggingClient.h:
* Source/WebCore/page/PerformanceMonitor.cpp:
* Source/WebCore/page/PerformanceMonitor.h:
* Source/WebCore/page/PerformanceObserverEntryList.cpp:
* Source/WebCore/page/PerformanceObserverEntryList.h:
* Source/WebCore/page/PerformanceUserTiming.cpp:
* Source/WebCore/page/PerformanceUserTiming.h:
* Source/WebCore/page/PointerCaptureController.cpp:
* Source/WebCore/page/PointerCaptureController.h:
* Source/WebCore/page/PointerLockController.cpp:
* Source/WebCore/page/PointerLockController.h:
* Source/WebCore/page/PrintContext.cpp:
* Source/WebCore/page/PrintContext.h:
* Source/WebCore/page/Quirks.cpp:
* Source/WebCore/page/Quirks.h:
* Source/WebCore/page/RemoteFrameClient.h:
* Source/WebCore/page/RemoteFrameView.cpp:
* Source/WebCore/page/RemoteFrameView.h:
* Source/WebCore/page/RenderingUpdateScheduler.cpp:
* Source/WebCore/page/RenderingUpdateScheduler.h:
* Source/WebCore/page/ResizeObserverEntry.h:
* Source/WebCore/page/ResourceUsageOverlay.cpp:
* Source/WebCore/page/ResourceUsageOverlay.h:
* Source/WebCore/page/SettingsBase.cpp:
* Source/WebCore/page/SettingsBase.h:
* Source/WebCore/page/SpeechRecognitionProvider.h:
* Source/WebCore/page/UserScript.cpp:
* Source/WebCore/page/UserScript.h:
* Source/WebCore/page/UserStyleSheet.cpp:
* Source/WebCore/page/UserStyleSheet.h:
* Source/WebCore/page/ValidationMessageClient.h:
* Source/WebCore/page/ViewportConfiguration.cpp:
* Source/WebCore/page/ViewportConfiguration.h:
* Source/WebCore/page/WheelEventDeltaFilter.cpp:
* Source/WebCore/page/WheelEventDeltaFilter.h:
* Source/WebCore/page/WorkerClient.h:
* Source/WebCore/page/cocoa/DataDetectionResultsStorage.h:
* Source/WebCore/page/csp/ContentSecurityPolicy.cpp:
* Source/WebCore/page/csp/ContentSecurityPolicy.h:
* Source/WebCore/page/csp/ContentSecurityPolicyDirective.cpp:
* Source/WebCore/page/csp/ContentSecurityPolicyDirective.h:
* Source/WebCore/page/csp/ContentSecurityPolicyDirectiveList.cpp:
* Source/WebCore/page/csp/ContentSecurityPolicyDirectiveList.h:
* Source/WebCore/page/csp/ContentSecurityPolicySource.cpp:
* Source/WebCore/page/csp/ContentSecurityPolicySource.h:
* Source/WebCore/page/ios/ContentChangeObserver.cpp:
* Source/WebCore/page/ios/ContentChangeObserver.h:
* Source/WebCore/page/ios/DOMTimerHoldingTank.cpp:
* Source/WebCore/page/ios/DOMTimerHoldingTank.h:
* Source/WebCore/page/linux/ResourceUsageOverlayLinux.cpp:
* Source/WebCore/page/mac/ServicesOverlayController.h:
* Source/WebCore/page/mac/ServicesOverlayController.mm:
* Source/WebCore/page/mac/TextIndicatorWindow.h:
* Source/WebCore/page/mac/TextIndicatorWindow.mm:
* Source/WebCore/page/mac/WheelEventDeltaFilterMac.h:
* Source/WebCore/page/mac/WheelEventDeltaFilterMac.mm:
* Source/WebCore/page/scrolling/ScrollAnchoringController.cpp:
* Source/WebCore/page/scrolling/ScrollAnchoringController.h:
* Source/WebCore/page/scrolling/ScrollLatchingController.cpp:
* Source/WebCore/page/scrolling/ScrollLatchingController.h:
* Source/WebCore/page/scrolling/ScrollingConstraints.cpp:
* Source/WebCore/page/scrolling/ScrollingConstraints.h:
* Source/WebCore/page/scrolling/ScrollingCoordinator.cpp:
* Source/WebCore/page/scrolling/ScrollingCoordinator.h:
* Source/WebCore/page/scrolling/ScrollingStateNode.cpp:
* Source/WebCore/page/scrolling/ScrollingStateNode.h:
* Source/WebCore/page/scrolling/ScrollingStateTree.cpp:
* Source/WebCore/page/scrolling/ScrollingStateTree.h:
* Source/WebCore/page/scrolling/ScrollingTree.cpp:
* Source/WebCore/page/scrolling/ScrollingTree.h:
* Source/WebCore/page/scrolling/ScrollingTreeNode.cpp:
* Source/WebCore/page/scrolling/ScrollingTreeNode.h:
* Source/WebCore/page/scrolling/ScrollingTreeScrollingNodeDelegate.cpp:
* Source/WebCore/page/scrolling/ScrollingTreeScrollingNodeDelegate.h:
* Source/WebCore/page/scrolling/mac/ScrollerPairMac.h:
* Source/WebCore/page/scrolling/mac/ScrollerPairMac.mm:
* Source/WebCore/page/writing-tools/WritingToolsController.h:
* Source/WebCore/page/writing-tools/WritingToolsController.mm:

Canonical link: <a href="https://commits.webkit.org/282576@main">https://commits.webkit.org/282576@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/486d2a845fbfd8924c7cc37814b8c3b00d251e49

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/63525 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/42881 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/16122 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/67546 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/14133 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/65645 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/50569 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/14413 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/51167 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/9778 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/66594 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/39811 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/55011 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/31844 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/36494 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/12382 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/13005 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/58004 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/12703 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/69242 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/7472 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/12286 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/58461 "Found 1 new test failure: fast/forms/search/search-size-with-decorations.html (failure)") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/7505 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/55094 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/58690 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/6246 "Passed tests") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/9612 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/38702 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/39781 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/40893 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/39524 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

dhecht commented on Source/WebCore/page/EditorClient.h:
```diff
@@ -35,6 +35,7 @@
 #include <wtf/CheckedPtr.h>
 #include <wtf/FastMalloc.h>
```
looks like you can remove FastMalloc.h

dhecht commented on Source/WebCore/page/ImageAnalysisQueue.h:
```diff
@@ -30,6 +30,7 @@
 #include "Timer.h"
 #include <wtf/FastMalloc.h>
```
remove?

msaboff replied:
Fixed.

dhecht commented on Source/WebCore/page/scrolling/ScrollingConstraints.cpp:
```diff
@@ -26,10 +26,14 @@
 #include "config.h"
 #include "ScrollingConstraints.h"
 
+#include <wtf/TZoneMalloc.h>
```
TZoneMallocInlines.h (I'm surprised it compiles...)

msaboff replied:
Fixed.

kmiller68 reviewed (approved):
r=me.

msaboff committed:
[TZone] WebCore/page Convert FastMalloc to TZone
https://bugs.webkit.org/show_bug.cgi?id=278419
rdar://134372652

Reviewed by Keith Miller.

Convert WTF_MAKE_FAST_ALLOCATED to WTF_MAKE_TZONE_ALLOCATED in
preparation for enabling TZone (not yet enabled).

* Source/WebCore/page/AlternativeTextClient.h:
* Source/WebCore/page/AttachmentElementClient.h:
* Source/WebCore/page/AudioOutputProvider.h:
* Source/WebCore/page/AutoscrollController.cpp:
* Source/WebCore/page/AutoscrollController.h:
* Source/WebCore/page/CaptionUserPreferences.cpp:
* Source/WebCore/page/CaptionUserPreferences.h:
* Source/WebCore/page/CaptionUserPreferencesMediaAF.cpp:
* Source/WebCore/page/CaptionUserPreferencesMediaAF.h:
* Source/WebCore/page/ContextMenuController.cpp:
* Source/WebCore/page/ContextMenuController.h:
* Source/WebCore/page/CryptoClient.h:
* Source/WebCore/page/DOMTimer.cpp:
* Source/WebCore/page/DOMTimer.h:
* Source/WebCore/page/DeviceClient.h:
* Source/WebCore/page/DeviceController.cpp:
* Source/WebCore/page/DeviceController.h:
* Source/WebCore/page/DiagnosticLoggingClient.h:
* Source/WebCore/page/DragClient.h:
* Source/WebCore/page/DragController.cpp:
* Source/WebCore/page/DragController.h:
* Source/WebCore/page/EditorClient.h:
* Source/WebCore/page/ElementTargetingController.cpp:
* Source/WebCore/page/ElementTargetingController.h:
* Source/WebCore/page/EmptyAttachmentElementClient.h:
* Source/WebCore/page/EventHandler.cpp:
* Source/WebCore/page/EventHandler.h:
* Source/WebCore/page/FocusController.cpp:
* Source/WebCore/page/FocusController.h:
* Source/WebCore/page/FrameView.cpp:
* Source/WebCore/page/FrameView.h:
* Source/WebCore/page/ImageAnalysisQueue.cpp:
* Source/WebCore/page/ImageAnalysisQueue.h:
* Source/WebCore/page/ImageOverlayController.cpp:
* Source/WebCore/page/ImageOverlayController.h:
* Source/WebCore/page/LocalFrameViewLayoutContext.cpp:
* Source/WebCore/page/LocalFrameViewLayoutContext.h:
* Source/WebCore/page/NavigatorLoginStatus.cpp:
* Source/WebCore/page/NavigatorLoginStatus.h:
* Source/WebCore/page/Page.cpp:
* Source/WebCore/page/Page.h:
* Source/WebCore/page/PageConfiguration.cpp:
* Source/WebCore/page/PageConfiguration.h:
* Source/WebCore/page/PageConsoleClient.cpp:
* Source/WebCore/page/PageConsoleClient.h:
* Source/WebCore/page/PageDebuggable.cpp:
* Source/WebCore/page/PageDebuggable.h:
* Source/WebCore/page/PageGroup.cpp:
* Source/WebCore/page/PageGroup.h:
* Source/WebCore/page/PageOverlay.cpp:
* Source/WebCore/page/PageOverlay.h:
* Source/WebCore/page/PageOverlayController.cpp:
* Source/WebCore/page/PageOverlayController.h:
* Source/WebCore/page/PagePasteboardContext.h:
* Source/WebCore/page/PerformanceLogging.cpp:
* Source/WebCore/page/PerformanceLogging.h:
* Source/WebCore/page/PerformanceLoggingClient.cpp:
* Source/WebCore/page/PerformanceLoggingClient.h:
* Source/WebCore/page/PerformanceMonitor.cpp:
* Source/WebCore/page/PerformanceMonitor.h:
* Source/WebCore/page/PerformanceObserverEntryList.cpp:
* Source/WebCore/page/PerformanceObserverEntryList.h:
* Source/WebCore/page/PerformanceUserTiming.cpp:
* Source/WebCore/page/PerformanceUserTiming.h:
* Source/WebCore/page/PointerCaptureController.cpp:
* Source/WebCore/page/PointerCaptureController.h:
* Source/WebCore/page/PointerLockController.cpp:
* Source/WebCore/page/PointerLockController.h:
* Source/WebCore/page/PrintContext.cpp:
* Source/WebCore/page/PrintContext.h:
* Source/WebCore/page/Quirks.cpp:
* Source/WebCore/page/Quirks.h:
* Source/WebCore/page/RemoteFrameClient.h:
* Source/WebCore/page/RemoteFrameView.cpp:
* Source/WebCore/page/RemoteFrameView.h:
* Source/WebCore/page/RenderingUpdateScheduler.cpp:
* Source/WebCore/page/RenderingUpdateScheduler.h:
* Source/WebCore/page/ResizeObserverEntry.h:
* Source/WebCore/page/ResourceUsageOverlay.cpp:
* Source/WebCore/page/ResourceUsageOverlay.h:
* Source/WebCore/page/SettingsBase.cpp:
* Source/WebCore/page/SettingsBase.h:
* Source/WebCore/page/SpeechRecognitionProvider.h:
* Source/WebCore/page/UserScript.cpp:
* Source/WebCore/page/UserScript.h:
* Source/WebCore/page/UserStyleSheet.cpp:
* Source/WebCore/page/UserStyleSheet.h:
* Source/WebCore/page/ValidationMessageClient.h:
* Source/WebCore/page/ViewportConfiguration.cpp:
* Source/WebCore/page/ViewportConfiguration.h:
* Source/WebCore/page/WheelEventDeltaFilter.cpp:
* Source/WebCore/page/WheelEventDeltaFilter.h:
* Source/WebCore/page/WorkerClient.h:
* Source/WebCore/page/cocoa/DataDetectionResultsStorage.h:
* Source/WebCore/page/csp/ContentSecurityPolicy.cpp:
* Source/WebCore/page/csp/ContentSecurityPolicy.h:
* Source/WebCore/page/csp/ContentSecurityPolicyDirective.cpp:
* Source/WebCore/page/csp/ContentSecurityPolicyDirective.h:
* Source/WebCore/page/csp/ContentSecurityPolicyDirectiveList.cpp:
* Source/WebCore/page/csp/ContentSecurityPolicyDirectiveList.h:
* Source/WebCore/page/csp/ContentSecurityPolicySource.cpp:
* Source/WebCore/page/csp/ContentSecurityPolicySource.h:
* Source/WebCore/page/ios/ContentChangeObserver.cpp:
* Source/WebCore/page/ios/ContentChangeObserver.h:
* Source/WebCore/page/ios/DOMTimerHoldingTank.cpp:
* Source/WebCore/page/ios/DOMTimerHoldingTank.h:
* Source/WebCore/page/linux/ResourceUsageOverlayLinux.cpp:
* Source/WebCore/page/mac/ServicesOverlayController.h:
* Source/WebCore/page/mac/ServicesOverlayController.mm:
* Source/WebCore/page/mac/TextIndicatorWindow.h:
* Source/WebCore/page/mac/TextIndicatorWindow.mm:
* Source/WebCore/page/mac/WheelEventDeltaFilterMac.h:
* Source/WebCore/page/mac/WheelEventDeltaFilterMac.mm:
* Source/WebCore/page/scrolling/ScrollAnchoringController.cpp:
* Source/WebCore/page/scrolling/ScrollAnchoringController.h:
* Source/WebCore/page/scrolling/ScrollLatchingController.cpp:
* Source/WebCore/page/scrolling/ScrollLatchingController.h:
* Source/WebCore/page/scrolling/ScrollingConstraints.cpp:
* Source/WebCore/page/scrolling/ScrollingConstraints.h:
* Source/WebCore/page/scrolling/ScrollingCoordinator.cpp:
* Source/WebCore/page/scrolling/ScrollingCoordinator.h:
* Source/WebCore/page/scrolling/ScrollingStateNode.cpp:
* Source/WebCore/page/scrolling/ScrollingStateNode.h:
* Source/WebCore/page/scrolling/ScrollingStateTree.cpp:
* Source/WebCore/page/scrolling/ScrollingStateTree.h:
* Source/WebCore/page/scrolling/ScrollingTree.cpp:
* Source/WebCore/page/scrolling/ScrollingTree.h:
* Source/WebCore/page/scrolling/ScrollingTreeNode.cpp:
* Source/WebCore/page/scrolling/ScrollingTreeNode.h:
* Source/WebCore/page/scrolling/ScrollingTreeScrollingNodeDelegate.cpp:
* Source/WebCore/page/scrolling/ScrollingTreeScrollingNodeDelegate.h:
* Source/WebCore/page/scrolling/mac/ScrollerPairMac.h:
* Source/WebCore/page/scrolling/mac/ScrollerPairMac.mm:
* Source/WebCore/page/writing-tools/WritingToolsController.h:
* Source/WebCore/page/writing-tools/WritingToolsController.mm:

Canonical link: https://commits.webkit.org/282576@main
